app/frontend/components/domains/requirements-library/requirement-field-edit/conditional-setup-modal.tsx

```diff
@@ -15,25 +18,119 @@ import {
   const { t } = useTranslation()
+
+  const formMethods = useFormContext<IRequirementBlockForm>()
+
+  const { control, watch, setValue } = formMethods
+
+  const watchedLabel = watch(`requirementsAttributes.${index}.label`)
+
+  const watchedWhen = watch(`requirementsAttributes.${index}.inputOptions.conditional.when`)
+  const watchedOperand = watch(`requirementsAttributes.${index}.inputOptions.conditional.operand`)
+  const watchedThen = watch(`requirementsAttributes.${index}.inputOptions.conditional.then`)
+  const allFieldsProvided = watchedWhen && watchedOperand && watchedThen
+
+  const watchedRequirements = watch(`requirementsAttributes`)
+  const watchedRequirementCode = watch(`requirementsAttributes.${index}.requirementCode`)
+  const selectedRequirementAttr = watchedRequirements?.find((reqAttr) => reqAttr.requirementCode === watchedWhen)
+
+  const [showAdvanced, setShowAdvanced] = useState<boolean>(false)
+  const toggleShowAdvanced = () => setShowAdvanced((cur) => !cur)
+
+  const [operandOptions, setOperandOptions] = useState<IOption[]>(null)
+  const [requirementOptions, setRequirementOptions] = useState<IOption[]>(null)
+
+  const onReset = () => {
+    setValue(`requirementsAttributes.${index}.inputOptions.conditional.when`, null)
+    setValue(`requirementsAttributes.${index}.inputOptions.conditional.operand`, null)
+    setValue(`requirementsAttributes.${index}.inputOptions.conditional.then`, null)
+  }
+
+  const getOperandOptions = () => {
+    if (selectedRequirementAttr?.inputOptions?.valueOptions) {
+      return selectedRequirementAttr?.inputOptions.valueOptions
+    }
+    if (selectedRequirementAttr?.inputType === "checkbox") {
+      return [
+        { label: t("ui.checked"), value: "true" },
+        { label: t("ui.unchecked"), value: "false" },
+      ]
+    }
+    return []
+  }
+
+  const getRequirementOptions = () => {
+    return watchedRequirements
+      .map((requirement) => ({ label: requirement.label, value: requirement.requirementCode }))
+      .filter((opt) => opt.value !== watchedRequirementCode)
+  }
+
+  useEffect(() => {
+    setRequirementOptions(getRequirementOptions())
+    if (watchedWhen) {
+      setOperandOptions(getOperandOptions())
+    }
+  }, [watchedWhen])
+
+  const effectOptions = ["show", "hide"].map((value) => ({
+    value,
+    // @ts-ignore
+    label: t(`requirementsLibrary.modals.conditionalSetup.${value}`),
+  }))
+
+  const getOperandSelectFormControl = (fieldName: keyof IRequirementBlockForm) => {
+    if (!watchedWhen) return <></>
+    const inputType = watchedRequirements.find((req) => req.requirementCode === watchedWhen).inputType
+    if (["text", "textarea", "phone", "email", "address", "bcaddress"].includes(inputType)) {
+      return <TextFormControl fieldName={fieldName} />
+    } else if (inputType === "number") {
+      return <NumberFormControl fieldName={fieldName} />
+    } else if (["select", "radio", "multi_option_select", "checkbox"].includes(inputType)) {
```

Comment:
not required, but would be nice to extract these to a constant variable, `["text", "textarea", "phone", "email", "address", "bcaddress"]`